src/base/server_type.cpp

```diff
@@ -93,6 +102,21 @@ int init_server_type (const char *db_name)
   return er_code;
 }
 
+void configure_single_node_server (SERVER_TYPE parameter_value)
+{
+  if (g_server_type != SERVER_TYPE_TRANSACTION || parameter_value != SERVER_TYPE_SINGLE_NODE)
+    {
+      return;
+    }
+
+  char *page_hosts_new_value;
+  page_hosts_new_value = (char *) malloc (22 * sizeof (char));
```

Comment:
Why do you choose the number '22'? Can I ask you a reason?

Reply:
If a "big enough" size is chosen, go for a power of two (e.g. 32) and give the number literal a name.
```c
  constexpr size_t PAGE_HOSTS_BUFSIZE = 32;
  char *page_hosts_new_value = (char *) malloc (PAGE_HOSTS_BUFSIZE); // will be freed by sysprm_final()
```

For a precise size, snprintf may be used to calculate the size (see [documentation](https://en.cppreference.com/w/c/io/fprintf))
```c
  const char *format = "localhost:%d";
  int portid = prm_get_master_port_id ();
  size_t bufsize = snprintf (NULL, 0, format, portid) + 1;
  char *page_hosts_new_value = (char *) malloc (bufsize); // will be freed by sysprm_final()
```

Reply:
Changed to a fix sized of 32, 16 will probably be enough as the portid is most likely a 4 digit number, but that will cause a  overflow warning from de compiler.